flowpipe/graph.py

```diff
@@ -228,19 +228,21 @@ def evaluate(self, mode="linear", skip_clean=False,
                 inputs have not changed since their output was computed
             submission_delay (float): The delay in seconds between loops
                 issuing new threads/processes if nodes are ready to process.
-            raise_after (int): The number of loops without currently running
-                threads/processes after which to raise a RuntimeError.
+            max_workers (int): The maximum number of parallel threads to spawn.
+                None defaults to your pythons ThreadPoolExecutor default.
         """
         log.info('Evaluating Graph "{0}"'.format(self.name))
 
+        # map mode keywords to evaluation functions and their arguments
         eval_modes = {
-            "linear": self._evaluate_linear,
-            "threading": self._evaluate_threaded,
-            "multiprocessing": self._evaluate_multiprocessed
+            "linear": (self._evaluate_linear, {}),
+            "threading": (self._evaluate_threaded, {"max_workers": max_workers}),
+            "multiprocessing": (self._evaluate_multiprocessed,
+                                {"submission_delay": submission_delay})
         }
 
         try:
-            eval_func = eval_modes[mode]
+            eval_func, eval_func_args = eval_modes[mode]
         except KeyError:
             mode_options = ""
```

Comment:
Not really part of the PR but I saw this recently, this could be solved more elegantly with join, like this:
```python
raise ValueError("Invalid mode {0}, options are {1}".format(
                mode, ", ".join(list(eval_modes.keys()))))
```